thoth_pkgdeps/image.py

```diff
@@ -116,6 +116,37 @@ def construct_rootfs(dir_path: str, rootfs_path: str) -> None:
                     os.remove(member.name)
                     tar_file.extract(member, set_attrs=False)
 
+        yield layer_digest
+
+
+def _analyzers_update_output(old_output: dict, layer: str,
+                             mercator_output: list, rpm_output: dict, rpm_repoquery_output: dict) -> dict:
+    old_output['mercator'] = old_output.get('mercator', [])
+    mercator_seen_packages = set()
+    for entry in old_output['mercator']:
+        mercator_seen_packages.add((entry['result'].get('name'), entry['result'].get('version'),
+                                    entry['digests']['manifest']))
+
+    for entry in mercator_output:
+        record = (entry['result']['name'], entry['result']['version'], entry['digests']['manifest'])
+        if record not in mercator_seen_packages:
+            entry['layer'] = layer
+            old_output['mercator'].append(entry)
+
+    old_output['rpm'] = old_output.get('rpm', [])
+    rpm_seen_packages = {entry['name'] for entry in old_output['rpm']}
+    for package in set(rpm_output) - rpm_seen_packages:
+        old_output['rpm'].append({
+            'name': package,
+            'layer': layer
+        })
+
+    old_output['rpm-dependencies'] = list(set(rpm_repoquery_output) | set(old_output.get('rpm-dependencies', set())))
```

Comment:
This is not sufficient, we need to keep track of actual dependencies (values present).